myriad-scheduler/src/main/java/org/apache/myriad/scheduler/TaskFactory.java

```diff
@@ -87,115 +76,19 @@ public NMTaskFactoryImpl(MyriadConfiguration cfg, TaskUtils taskUtils, ExecutorC
-    CommandInfo getCommandInfo(ServiceResourceProfile profile, NMPorts ports) {
-      MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
-      CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
-      String cmd;
-
-      if (myriadExecutorConfiguration.getJvmUri().isPresent()) {
-        final String jvmRemoteUri = myriadExecutorConfiguration.getJvmUri().get();
-        LOGGER.info("Getting JRE distribution from:" + jvmRemoteUri);
-        URI jvmUri = URI.newBuilder().setValue(jvmRemoteUri).setExtract(true).build();
-        commandInfo.addUris(jvmUri);
-      }
-
-      if (myriadExecutorConfiguration.getConfigUri().isPresent()) {
-        String configURI = myriadExecutorConfiguration.getConfigUri().get();
-        LOGGER.info("Getting Hadoop distribution from: {}", configURI);
-        commandInfo.addUris(URI.newBuilder().setValue(configURI).build());
-      }
-
-      if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
-        //Both FrameworkUser and FrameworkSuperuser to get all of the directory permissions correct.
-        if (!(cfg.getFrameworkUser().isPresent() && cfg.getFrameworkSuperUser().isPresent())) {
-          throw new RuntimeException("Trying to use remote distribution, but frameworkUser" + "and/or frameworkSuperUser not set!");
-        }
-        String nodeManagerUri = myriadExecutorConfiguration.getNodeManagerUri().get();
-        cmd = clGenerator.generateCommandLine(profile, ports);
-
-        //get the nodemanagerURI
-        //We're going to extract ourselves, so setExtract is false
-        LOGGER.info("Getting Hadoop distribution from: {}", nodeManagerUri);
-        URI nmUri = URI.newBuilder().setValue(nodeManagerUri).setExtract(false).build();
-
-        //get configs directly from resource manager
-        String configUrlString = clGenerator.getConfigurationUrl();
-        LOGGER.info("Getting config from:" + configUrlString);
-        URI configUri = URI.newBuilder().setValue(configUrlString).build();
-        LOGGER.info("Slave will execute command: {}",  cmd);
-        commandInfo.addUris(nmUri).addUris(configUri).setValue("echo \"" + cmd + "\";" + cmd);
-        commandInfo.setUser(cfg.getFrameworkSuperUser().get());
-
-      } else {
-        cmd = clGenerator.generateCommandLine(profile, ports);
-        commandInfo.setValue("echo \"" + cmd + "\";" + cmd);
-
-        if (cfg.getFrameworkUser().isPresent()) {
-          commandInfo.setUser(cfg.getFrameworkUser().get());
-        }
-      }
-      return commandInfo.build();
-    }
-
     @Override
     public TaskInfo createTask(Offer offer, FrameworkID frameworkId, TaskID taskId, NodeTask nodeTask) {
       Objects.requireNonNull(offer, "Offer should be non-null");
       Objects.requireNonNull(nodeTask, "NodeTask should be non-null");
 
-      NMPorts ports = getPorts(offer);
+      AbstractPorts ports = taskUtils.getRandomPortResources(offer, 4, new TreeSet<Long>());
```

Comment:
I want the move it out, stay tuned.  There's more work to do in the TaskFactory Code and ResourceOfferEventHandler, I'll then get rid of a lot of TaskUtils methods.